src/main/scala/com/gu/fulfilmentLookup/AwsS3Client.scala

```diff
@@ -0,0 +1,37 @@
+package com.gu.fulfilmentLookup
+
+import java.io.InputStream
+import com.amazonaws.services.s3.AmazonS3ClientBuilder
+import com.amazonaws.services.s3.model.GetObjectRequest
+import purecsv.unsafe.CSVReader
+import scala.io.Source
+import scala.util.Try
+
+trait CsvClient {
+  def getDeliveryRows(s3Stream: InputStream): Try[List[DeliveryRow]]
+  def getDeliveryRowsFromS3(bucketName: String, subFolder: String, fileName: String): Try[List[DeliveryRow]]
+}
+
+object AwsS3Client extends CsvClient with Logging {
+
+  val s3Client = AmazonS3ClientBuilder.defaultClient()
+
+  override def getDeliveryRows(s3Stream: InputStream): Try[List[DeliveryRow]] = {
+    val csvAsString = Source.fromInputStream(s3Stream).mkString
```

Comment:
it looks like mkString can throw an IOException so it mayb e worth wrapping it in the try too